### ISSUE TITLE
selz.co shortlinks require two clicks to open in mobile Safari 

### ISSUE THREAD

**Issue by cwhenderson20** [Selz/jquery.selz#5]
Tested and confirmed in iOS 8 on iPhone and iPad. Invariably, the first click does nothing, and the second click opens the modal in a new tab as expected. After closing the modal/window that opens, all subsequent clicks of the same link will work; other selz.co shortlinks on the same page (i.e., those pointing to different products) will require the same two-click pattern to get the window to open.

In all other browsers that I have tested (Chrome for iOS, Chrome for Mac, Chrome for Android (5.0.4), Safari for Mac, Firefox for Mac), only one click is required to correctly open the modal/window.

I am using jquery.selz.min.js provided by the CDN; the issue is happening with both 1.0.5 and 1.0.6 versions of the script, as well as with both the latest 1.x and 2.x versions of jQuery.

**Comment by SamPotts** [comment]
Hey @cwhenderson20,

Sorry I totally missed this. Is this still an issue? Perhaps it's related to the issue you just raised in #6 ? I'm about to push a fix for that (using `this` as you suggested) so could you re-test? :+1:

**Comment by cwhenderson20** [comment]
@SamPotts, it's not a problem. Yes, it's still happening. My current workaround to add a listener to all Selz links, detect if the user is on mobile safari, and if so, set a timeout for a second click 500 milliseconds later (I found that 0, 1, etc. were all too small and would work _sometimes_ but not all of the time).

```javascript
var ua = navigator.userAgent;
var iOS = !!ua.match(/iPad/i) || !!ua.match(/iPhone/i);
var webkit = !!ua.match(/WebKit/i);
var iOSSafari = iOS && webkit && !ua.match(/CriOS/i);

$('.js-selzLink').one('click', function (event) {
        if (iOSSafari) {
            setTimeout(function () {
                $(event.target).click();
            }, 500);
        } else {
            return;
        }
    });
```

I did start to dig into the problem when I first noticed it, but I didn't figure out exactly what was different between mobile Safari and all of the rest of the browsers.

**Comment by cwhenderson20** [comment]
Apologies @SamPotts, this actually is fixed in 1.0.7. I had forgotten to remove my local, modified copy of the file while testing.

**Comment by SamPotts** [comment]
Awesome! I guess it was all related the event bubbling in #6 .

Thanks for your help testing etc. If there's anything else, don't hesitate to give us a yell! :+1:

**Comment by cwhenderson20** [comment]
No problem. I just found it strange that it worked in mobile Chrome, seeing as it's just a wrapper around WebKit.

Hopefully there won't be anything else, but I'll be on the lookout :+1:

**Comment by cwhenderson20** [comment]
Hate to have to do this, but I'm afraid this issue is not actually fixed. Again, my mistake as not only had I modified a local copy of the file, but I was also referencing the CDN version, which meant each tap was actually being dealt with twice. 

After confirming multiple times that I was referencing only the hosted 1.0.7 js file, I can say that shortlinks still require two clicks to open in mobile Safari. :(

**Comment by cwhenderson20** [comment]
It actually appears that turning the `prefetch` option to `true` in the `$.selz` initialization fixes this for all browsers. Must be a mobile Safari timing issue. I'll keep this open in case you still consider it a bug, but I'm happy just keeping the option for prefetching on. It lets me remove lots of hacky-looking code :+1:

Otherwise, as far as I can tell, the first click always gets "stuck" in the `else` block of the `openOverlay` function that grabs the item data if it doesn't already exist. Then, since the `modalUrl` has already been attached after the first click, the second click triggers the `if` block's `window._$elz.m.open`. What I can't figure out is why they're different — they're basically identical calls.

```javascript
if (typeof modalUrl === "string" && modalUrl.length > 0) {
    // window *always* opens here
    window._$elz.m.open(modalUrl, null);
} else {
    getItemData($trigger, function(data) {
        // window *never* opens here
        window._$elz.m.open(data.Url, null);
    });
}
```

**Comment by SamPotts** [comment]
Hey mate, I've done some more digging and think I've worked out what the issue was. Because of issues with `position:fixed` on some mobile browsers (in particular iOS), we opt to open the item in a new tab/window on those browsers using `window.open()`. This obviously needs to be user initiated rather than programatically. The issue was we were fetching the item data with AJAX and then calling `window.open()` which obviously didn't work but explains why it worked the 2nd time, because we cache the data, so there was no need for the request that 2nd time. 

To fix it, I've made the plugin *always* prefetch data. This has pros and negatives of course. On the one side, it's another request (very small) but on the plus, it means it'll be faster for the user to initiate the overlay when they choose to. Personally, the later outweighs the former in terms of experience.

Thanks again for your help with this! :+1:
